src/main/java/org/junitpioneer/jupiter/StdIoExtension.java

```diff
@@ -92,6 +95,21 @@ private void swapAndStoreIn(ExtensionContext context, StdIn stdIn) {
 		System.setIn(stdIn); //NOSONAR required to redirect output
 	}
 
+	private StdErr prepareStdErr(ExtensionContext context) {
```

Comment:
Maye we can generalize/reuse the `prepare`, `store`, and `create` methods for `StdOut` and `StdErr`, using the new abstract base class `StdIoStream`?

Reply:
Not part of the behaviour, this logic belongs to the extension itself. I'm open to refactoring it to avoid duplication, but I don't think that's the right approach.